kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl.java

```diff
@@ -82,7 +82,42 @@
 
   @Override
   public List<HasMetadata> waitUntilReady(final long amount, final TimeUnit timeUnit) throws InterruptedException {
-    return waitUntilCondition(Objects::nonNull, amount, timeUnit);
+    List<HasMetadata> items = acceptVisitors(asHasMetadata(item, true), visitors);
+    if (items.size() == 0) {
+      return Collections.emptyList();
+    }
+
+    final List<HasMetadata> result = new ArrayList<>();
+    final List<HasMetadata> itemsWithConditionNotMatched = new ArrayList<>(items);
+    final int size = items.size();
+    final ExecutorService executor = Executors.newFixedThreadPool(size);
+
+    try {
+      final CountDownLatch latch = new CountDownLatch(size);
+      for (final HasMetadata meta : items) {
+        final ResourceHandler<HasMetadata, HasMetadataVisitiableBuilder> h = handlerOf(meta);
+        executor.submit(() -> {
+          try {
+            result.add(h.waitUntilReady(client, config, meta.getMetadata().getNamespace(), meta, amount, timeUnit));
+          } catch (Throwable t) {
+            //consider all errors as not ready.
```

Comment:
This is tricky.

If a resource doesn't support the notion of waiting `Readiness.isReady` will throw an Exception, so with the current approach, you will wait until timeout. For that particular case, we should do exactly the opposite `consider resource ready`.